provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/ProcessorBlackDuckServicesFactoryCache.java

```diff
@@ -0,0 +1,87 @@
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.collections4.map.PassiveExpiringMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.exception.AlertConfigurationException;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.provider.blackduck.BlackDuckProperties;
+import com.synopsys.integration.alert.provider.blackduck.factory.BlackDuckPropertiesFactory;
+import com.synopsys.integration.blackduck.http.client.BlackDuckHttpClient;
+import com.synopsys.integration.blackduck.service.BlackDuckServicesFactory;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+/**
+ * Caches {@link BlackDuckServicesFactory} by BlackDuck configuration ID.
+ * Synchronized and backed by a {@link PassiveExpiringMap} the documentation for which
+ * can be found <a href="https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/map/PassiveExpiringMap.html">here</a>.
+ */
+@Component
+public class ProcessorBlackDuckServicesFactoryCache {
+    public static final long MAX_TIME_TO_LIVE_MINUTES = 2;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final BlackDuckPropertiesFactory blackDuckPropertiesFactory;
+    private final Map<Long, BlackDuckServicesFactory> servicesFactoryCache;
+
+    public ProcessorBlackDuckServicesFactoryCache(BlackDuckPropertiesFactory blackDuckPropertiesFactory) {
+        this.blackDuckPropertiesFactory = blackDuckPropertiesFactory;
+        this.servicesFactoryCache = Collections.synchronizedMap(new PassiveExpiringMap<>(MAX_TIME_TO_LIVE_MINUTES, TimeUnit.MINUTES));
+    }
+
+    public BlackDuckServicesFactory retrieveBlackDuckServicesFactory(Long blackDuckConfigId) throws AlertConfigurationException {
+        BlackDuckServicesFactory blackDuckServicesFactory = servicesFactoryCache.get(blackDuckConfigId);
+        if (null == blackDuckServicesFactory) {
+            blackDuckServicesFactory = createBlackDuckServicesFactory(blackDuckConfigId);
+        }
+
+        servicesFactoryCache.put(blackDuckConfigId, blackDuckServicesFactory);
+        return blackDuckServicesFactory;
+    }
+
+    private BlackDuckServicesFactory createBlackDuckServicesFactory(Long blackDuckConfigId) throws AlertConfigurationException {
+        Optional<BlackDuckProperties> optionalProperties = blackDuckPropertiesFactory.createPropertiesIfConfigExists(blackDuckConfigId);
+        if (optionalProperties.isPresent()) {
+            BlackDuckProperties blackDuckProperties = optionalProperties.get();
+            Slf4jIntLogger intLogger = new Slf4jIntLogger(logger);
+            try {
+                BlackDuckHttpClient blackDuckHttpClient = blackDuckProperties.createBlackDuckHttpClient(intLogger);
```

Comment:
Shouldn't this be using the `blackDuckProperties.createBlackDuckCacheClient()` method, so that the requests to BlackDuck can be cached?